src/rocksdb/src/lib.rs

```diff
@@ -890,12 +879,23 @@ fn rocksdb_core_loop<K, V, M, O, IM, F>(
                     }
                     Err(e) => response_sender.send(Err(e)),
                 };
+
+                // TODO(maz): should we also terminate when retries are exhausted?
+                if let Some(fatal_error) = fatal_error {
```

Comment:
I couldn't figure out why we track this error separately instead of inspecting the `retry_result` above. Do we lose some information through that variable?